api/v1beta1/kokumetricsconfig_types.go

```diff
@@ -0,0 +1,476 @@
+
+	// Token allows upload of data using token authentication.
+	Token AuthenticationType = "token"
+)
+
+// EmbeddedObjectMetadata contains a subset of the fields included in k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta
+// Only fields which are relevant to embedded resources are included.
+type EmbeddedObjectMetadata struct {
+	// Name must be unique within a namespace. Is required when creating resources, although
+	// some resources may allow a client to request the generation of an appropriate name
+	// automatically. Name is primarily intended for creation idempotence and configuration
+	// definition.
+	// Cannot be updated.
+	// More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	// +optional
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+
+	// Map of string keys and values that can be used to organize and categorize
+	// (scope and select) objects. May match selectors of replication controllers
+	// and services.
+	// More info: http://kubernetes.io/docs/user-guide/labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty" protobuf:"bytes,11,rep,name=labels"`
+
+	// Annotations is an unstructured key value map stored with a resource that may be
+	// set by external tools to store and retrieve arbitrary metadata. They are not
+	// queryable and should be preserved when modifying objects.
+	// More info: http://kubernetes.io/docs/user-guide/annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,12,rep,name=annotations"`
+}
+
+// EmbeddedPersistentVolumeClaim is an embedded version of k8s.io/api/core/v1.PersistentVolumeClaim.
+// It contains TypeMeta and a reduced ObjectMeta.
+type EmbeddedPersistentVolumeClaim struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// EmbeddedMetadata contains metadata relevant to an EmbeddedResource.
+	EmbeddedObjectMetadata `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines the desired characteristics of a volume requested by a pod author.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#persistentvolumeclaims
+	// +optional
+	Spec corev1.PersistentVolumeClaimSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// AuthenticationSpec defines the desired state of Authentication object in the KokuMetricsConfigSpec.
+type AuthenticationSpec struct {
+
+	// AuthType is a field of KokuMetricsConfig to represent the authentication type to be used basic or token.
+	// Valid values are:
+	// - "basic" : Enables authentication using user and password from authentication secret.
+	// - "token" (default): Uses cluster token for authentication.
+	// +kubebuilder:default="token"
+	AuthType AuthenticationType `json:"type"`
+
+	// AuthenticationSecretName is a field of KokuMetricsConfig to represent the secret with the user and password used for uploads.
+	// +optional
+	AuthenticationSecretName string `json:"secret_name,omitempty"`
```

Comment:
Previously we discussed transitioning the underscores `secret_name` to camel case `secretName` when we went to a new CRD version. Do we still want to do that?